db/cache/charts.go

```diff
@@ -1254,13 +1254,13 @@ func dailyHashrate(time, chainwork ChartUints) (ChartUints, ChartUints) {
 	rates := make([]uint64, 0, len(time)-1)
 	var dupes int
 	for i, t := range time[1:] {
-		tDiff := t - time[i]
+		tDiff := int64(t - time[i])
 		if tDiff <= 0 {
 			tDiff = aDay
 			dupes += 1
 		}
 		workDiff := chainwork[i+1] - chainwork[i]
-		rates = append(rates, (workDiff)*1e6/tDiff)
+		rates = append(rates, (workDiff)*1e6/uint64(tDiff))
```

Comment:
@buck54321 Would you have a look at this?  Any gotchas?  This seems to be a solid bugfix so `tDiff <= 0` can work.

Reply:
Oh yeah. Good catch.

Reply:
Merged separately in https://github.com/decred/dcrdata/pull/1589